### PR TITLE
Benchmarks for model builders from XGBoost and LightGBM models

### PR DESCRIPTION
**Pull request by RukhovichIV**

### CODE REVIEW

**Comment by RukhovichIV** [comment]
@PetrovKP, @Alexsandruss, @ShvetsKS,
Added LightGBM and XGBoost benchmarks as a new "lib".
I tried to use common functions from other libs (e. g. from `./xgboost/bench.py`), but they're unavailable from running scope, so I made a new file (`./modelbuilders/bench.py`), which consists of copied, slightly shortened, minimum required functions from `./xgboost/bench.py`. I only added my `get_accuracy()` function, because it's much shorter, than what I found there.
I also pushed a MR with new configs to https://gitlab.devtools.intel.com/aandree1/sklbench_utils/-/merge_requests/5, it works great together.
I also added `single-precision-histogram` and `enable-experimental-json-serialization` parameters to xgboost benchmarks

File: .gitignore [hunk 1]
```diff
@@ -11,3 +11,4 @@ __work*
 # Datasets
 dataset
 *.csv
+*.npy
```
Comment (PetrovKP) [inline .gitignore:14]: Add EOF

File: modelbuilders/bench.py [hunk 2]
```diff
@@ -0,0 +1,509 @@
+import argparse
```
Comment (PetrovKP) [inline modelbuilders/bench.py:1]: Copyright ?

File: modelbuilders/lgbm_mb.py [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+    test_metric_daal = metric_func(y_test, daal_pred.prediction)
+
+print_output(library='modelbuilders', algorithm=f'lightgbm_{task}_and_modelbuilder',
+             stages=['lgbm_train_matrix_create', 'lgbm_test_matrix_create', 'lgbm_training',
+                'lgbm_prediction', 'lgbm_to_daal_conv', 'daal_prediction'],
+             columns=columns, params=params, functions=['lgbm_dataset', 'lgbm_dataset', 'lgbm_train',
+                'lgbm_predict', 'lgbm_to_daal', 'daal_compute'],
+             times=[t_creat_train, t_creat_test, t_train, t_lgbm_pred, t_trans, t_daal_pred],
+             accuracy_type=metric_name, accuracies=[0, 0, train_metric, test_metric_xgb, 0, test_metric_daal],
+             data=[X_train, X_test, X_train, X_test, X_train, X_test])
```
Comment (PetrovKP) [inline modelbuilders/lgbm_mb.py:130]: eof

File: modelbuilders/bench.py [hunk 4]
```diff
@@ -0,0 +1,509 @@
...
+import numpy as np
+import sklearn
+import timeit
+import json
+
+
+def columnwise_score(y, yp, score_func):
```
Comment (Alexsandruss) [inline modelbuilders/bench.py:8]: bench.py file should be similar in all folder (sklearn, daal4py, etc.)

File: modelbuilders/lgbm_mb.py (hunk 3 above)
Comment (Alexsandruss) [inline modelbuilders/lgbm_mb.py:130]: Add newline

File: modelbuilders/xgb_mb.py [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+    t_daal_pred, daal_pred = measure_function_time(predict_algo.compute, X_test, model_daal, params=params)
+    test_metric_daal = metric_func(y_test, daal_pred.prediction)
+else:
+    predict_algo = daal4py.gbt_regression_prediction()
+    t_daal_pred, daal_pred = measure_function_time(predict_algo.compute, X_test, model_daal, params=params)
+    test_metric_daal = metric_func(y_test, daal_pred.prediction)
+
+print_output(library='modelbuilders', algorithm=f'xgboost_{task}_and_modelbuilder',
+             stages=['xgb_train_dmatrix_create', 'xgb_test_dmatrix_create', 'xgb_training', 'xgb_prediction', 
+                'xgb_to_daal_conv', 'daal_prediction'],
```
Comment (Alexsandruss) [inline modelbuilders/xgb_mb.py:148]: Use flake8 to correct formatting:
`pip/conda install flake8`
`flake8 <folder or file names>`
You can ignore 'too long line' if line is complicated

File: xgboost/gbt.py [hunk 6]
```diff
@@ -65,6 +65,12 @@ def convert_xgb_predictions(y_pred, objective):
                     choices=('reg:squarederror', 'binary:logistic',
                              'multi:softmax', 'multi:softprob'),
                     help='Control a balance of positive and negative weights')
+parser.add_argument('--count-dmatrix', default=False, action='store_true',
+                    help='Count DMatrix creation in time measurements')
+parser.add_argument('--single-precision-histogram', default=False, action='store_true',
+                    help='Build histograms instead of double precision')
+parser.add_argument('--enable-experimental-json-serialization', default=True,
```
Comment (Alexsandruss) [inline xgboost/gbt.py:72]: `default=False` is better if this feature affects perf.

**Review by Alexsandruss — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 1 above)
Comment (RukhovichIV) [reply .gitignore]: done

File: modelbuilders/bench.py (hunk 2 above)
Comment (RukhovichIV) [reply modelbuilders/bench.py:6]: added

File: modelbuilders/lgbm_mb.py (hunk 3 above)
Comment (RukhovichIV) [reply modelbuilders/lgbm_mb.py]: added too

File: modelbuilders/bench.py (hunk 4 above)
Comment (RukhovichIV) [reply modelbuilders/bench.py]: Thank you. Added same `bench.py` as in other folders.
I also added the `utils.py` file with function, that I use in both my benchmarks

File: modelbuilders/lgbm_mb.py (hunk 3 above)
Comment (RukhovichIV) [reply modelbuilders/lgbm_mb.py]: done it

File: modelbuilders/xgb_mb.py (hunk 5 above)
Comment (RukhovichIV) [reply modelbuilders/xgb_mb.py]: done it for every file, which I added / changed

Comment (RukhovichIV) [reply modelbuilders/xgb_mb.py]: I also used autopep8 formatting, so every line is less or equal 100 characters now

File: xgboost/gbt.py (hunk 6 above)
Comment (RukhovichIV) [reply xgboost/gbt.py:72]: It's true by default in XGBoost, so, as we discussed, I decided to leave it as is

**Review by Alexsandruss — APPROVED** [review]

**Pull request merged**